impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -227,13 +227,10 @@ public OnHeapStore(final Configuration<K, V> config, final TimeSource timeSource
     this.keyType = config.getKeyType();
     this.valueType = config.getValueType();
     this.expiry = config.getExpiry();
+    this.keyCopier = keyCopier;
```

Comment:
Could we eliminate the keyCopier here by having the backend return another backend on clear?  Is that cleaner?